### PR TITLE
Stop using old serialization in `Blueprint`

### PR DESCRIPTION
**Pull request by RiscadoA**
# Description

Changes `Blueprint` to use only reflection and none of the old serialization code.
Changes `Commands` to support the new `Blueprint` interface.
Changes component storages so that type-erased components can be added to them.

## Checklist

- [x] Self-review changes.
- [x] Evaluate impact on the documentation.
- [x] Ensure test coverage.
- [x] ~~Write new samples.~~

### CODE REVIEW

File: engine/src/cubos/engine/tools/scene_editor/plugin.cpp [hunk 1]
```diff
@@ -81,7 +81,7 @@ static void openScene(const Asset<Scene>& sceneToSpawn, Commands& commands, cons
     closeScene(commands, scene);
     auto sceneRead = assets.read(sceneToSpawn);
     auto builder = commands.spawn(sceneRead->blueprint);
-    for (const auto& [entity, name] : sceneRead->blueprint.getMap())
+    for (const auto& [entity, name] : sceneRead->blueprint.bimap())
```
Comment (RiscadoA) [inline engine/src/cubos/engine/tools/scene_editor/plugin.cpp:84]: @DiogoMendonc-a you were picked up by the code owners stuff only because of this change. If you don't want to review the rest its okay :laughing:

**Comment by RiscadoA** [comment]
If this becomes hard to follow for anyone who is reviewing this, lmk and I can guide you through the changes.

File: core/include/cubos/core/ecs/blueprint.hpp [hunk 2]
```diff
@@ -4,210 +4,135 @@
...
 #include <cubos/core/memory/type_map.hpp>
+#include <cubos/core/memory/unordered_bimap.hpp>
 
 namespace cubos::core::ecs
 {
-    /// @brief Stores a bundle of entities and their respective components, which can be easily
-    /// spawned into a world. This is in a way the 'Prefab' of @b CUBOS., but lower level.
+    /// @brief Collection of entities and their respective components.
+    ///
+    /// Blueprints are in a way the 'Prefab' of @b CUBOS. They act as tiny @ref World which can
```
Comment (RiscadoA) [inline core/include/cubos/core/ecs/blueprint.hpp:20]:
```suggestion
    /// Blueprints are in a way the 'Prefab' of @b CUBOS. They act as a tiny @ref World which can
```

**Review by DiogoMendonc-a — APPROVED** [review]

**Review by roby2014 — APPROVED** [review]
